### PR TITLE
[SPARK-35788][SS] Metrics support for RocksDB instance

### PR DESCRIPTION
**Pull request by xuanyuanking**
### What changes were proposed in this pull request?
Add more metrics for the RocksDB instance. We transform the native states from RocksDB.

### Why are the changes needed?
Improve the usability with more metrics for RocksDB instance.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
retest this please

**Review by viirya — COMMENTED** [review]
I don't find the different change than other related PRs. @xuanyuanking Which is different change here?

**Comment by xuanyuanking** [comment]
Sorry @viirya, my mistake. Just push the latest commit of the metrics support.

**Comment by HeartSaVioR** [comment]
Shall we rebase this one for continue reviewing?

**Comment by HeartSaVioR** [comment]
Same rebasing issue happens here as well.

Could we please do the "clean" rebase? Like below:

```
git checkout master
git pull
git checkout -b SPARK-35788-v2
git cherry-pick 7b6da3d (and following commits)
git log <= note the latest commit ID (say, "commit_id")
git checkout SPARK-35788
git reset --hard "commit_id"
git push --force
```

**Comment by xuanyuanking** [comment]
Thanks for the guidance, let me try.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala:523]: nit: indent

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala:150]: nit: latest`L`oadCheckpointMetrics

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala:481]: nit: indent

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala:500]: nit: indent

**Review by HeartSaVioR — COMMENTED** [review]
Looks good overall. Only nits.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala
Comment (xuanyuanking) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala]: Thanks, done in 355953e.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala
Comment (xuanyuanking) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala]: Thanks, done in 355953e

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala
Comment (xuanyuanking) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala]: Thanks, done in 355953e

**Comment by xuanyuanking** [comment]
retest this please

**Review by HeartSaVioR — APPROVED** [review]
+1

I'll wait for @viirya to take a look; I'll merge this in a couple of days if there's no comment.

**Comment by viirya** [comment]
I'm in vacation, but I will find some time looking at this. Thanks.

**Comment by HeartSaVioR** [comment]
Thanks for the update. Let's just merge this in and add up the reviews in the next PR then. It should be pretty much simpler as we will have only one PR for RocksDB state store after merging this.

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by viirya** [comment]
Okay for me. Thanks!

**Comment by HeartSaVioR** [comment]
Thanks @xuanyuanking for the contribution! I merged this to master/branch-3.2.

Please rebase the next PR.

**Pull request closed without merging**
